### DISCUSSION TITLE
Scrollytelling and Page Formatting

### DISCUSSION THREAD

**Discussion by ThiernoMbengue** [quarto-dev/quarto-cli#7960] [Q&A]
### Description

Hello,

I am currently working on a Quarto project with two main objectives. The first is to develop a data visualization website similar to this one: https://vizagreste.agriculture.gouv.fr/. The site will include various charts and features, among which are:

Scrollytelling: For example, on the site https://vizagreste.agriculture.gouv.fr/evolution-du-nombre-d-exploitations.html, there is text that scrolls with the mouse, and also a diagram that changes color in accordance with the displayed explanations.

Page Formatting: The goal is to make the page very readable and aesthetically pleasing.

The second objective is to simplify this process as much as possible by using Quarto to automate visualization with various types of data. The aim is to make these tools easily accessible to individuals with basic R programming skills, enabling them to easily integrate their data and create visualizations.

My questions are as follows: Is it possible to create a website with these features using Quarto, while maintaining a simple approach? Specifically, is it possible to utilize the main scrollytelling options available in Quarto and optimize the web page layout? Do you have any suggestions or ideas that could help me in realizing this project?

Thank you in advance for your assistance.

**Comment by kshitizkhanal7** [comment]
It's not impossible but I suppose only a few people are working on it. 
Some decent examples here: https://jimjam-slam.github.io/quarto-experiments/scrollytelling/ This uses scrollama.js
I haven't tried much. One way is to integrate Shiny apps or HTML widgets, but that's more for interactivity than scrollies.

**Reply by ThiernoMbengue** [reply]
Thank you for your response, I will check the link you provided, thanks.

**Reply by jimjam-slam** [reply]
@andrewpbray and I are working on a more comprehensive solution for scrollytelling in Quarto! Stay tuned!

**Reply by ThiernoMbengue** [reply]
Okay I will pay attention to that, thank you

**Reply by ThiernoMbengue** [reply]
Hello, the last time we spoke, you told me that you were working on a more comprehensive solution for scrollytelling in Quarto... Do you have any updates on that, or do you have other proposals for scrollytelling on Quarto?

**Reply by jimjam-slam** [reply]
It's still a work in progress, I'm afraid! We're on the home stretch, but I'm not sure that we're quite ready to start getting public feedback on it.

If it's something you're interested in pursuing yourself, you could try using a JavaScript library like [scrollama.js](https://github.com/russellsamora/scrollama) (the library I use above and that we're integrating ourselves), or tinker with the upcoming [scroll view timelines](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-timeline#browser_compatibility) in CSS (available in very new versions of Edge and Chrome, but not in Firefox or Safari yet... so not many people).

**Reply by ThiernoMbengue** [reply]
Thank you for your response and your suggestions

**Reply by declann** [reply]
Hi! I used this link to start experimenting with some scrollytelling - really useful and nice to see working right within Quarto!, and finally I released a blog with this interaction technique. It's here if anyones interested: https://calcwithdec.dev/posts/viz-risk/ and source is linked, but it's a mess since I was trying a lot of things that are new for me! (especially a vega lite visualization with a lot of param/signal things going on)

scrollama code is near the bottom. `resize` generates a lot of events that were tricky to cope with (and triggered more often than I anticipated: the way mobile hides browser navigation etc.)
But I guess I'll get more comfortable with a few of these things next time.

**Reply by jimjam-slam** [reply]
This is really good, @declann! 👏🏻

**Reply by declann** [reply]
Thanks for the template to get me looking at this @jimjam-slam !
I thought `vh` units were super convenient making that but now I realise from actually reading scrollama docs that they were part to blame for my `resize` complaints

Anyways, I think the main point in these experiments is that Quarto isn't much getting in the way of creating these scroller type experiences.

**Reply by jimjam-slam** [reply]
Oh, that's good to know, @declann! Handling resizing has been a bit of a pain for @andrewpbray and I too extension. It's been a while since we looked at the spacing between narrative blocks; we'll take another look at it!

**Reply by declann** [reply]
Cool!

On resize then:
There's a resize trigger in scrollama, that I link to `window.onresize` events in a kinda elaborate way [here](https://gitlab.com/declann/calcwithdec.dev/-/blob/main/posts/viz-risk/index.qmd?ref_type=heads#L1529)

It's a mess what I did, but hopefully will find that a lot of it was unnecessary! I need to study scrollama more.

When I think about it now, maybe when the trigger is used scrollama tries to resync state: by rerunning steps from index 0. I saw these events and inhibited them instead. I rely on state updates in progress and in step changes but maybe more state changes should happen in step change even when it 'already' happens on scroll through progress, so that it can be recreated in this way too.

For spacing feel free to check my (familiar!) [html pattern](https://gitlab.com/declann/calcwithdec.dev/-/blob/main/posts/viz-risk/index.qmd?ref_type=heads#L111) (also [scroller.setup](https://gitlab.com/declann/calcwithdec.dev/-/blob/main/posts/viz-risk/index.qmd?ref_type=heads#L1534) is here) and if ye make something cool please share :D

Scrollama docs has [this caveat](https://github.com/russellsamora/scrollama?tab=readme-ov-file#scrollamaresize) about the resize trigger, but somehow I thought it was necessary in what I setup. If you don't hook that trigger up to window.onresize like I did, then maybe the weird event re-runs won't happen.

**Reply by fsemecurbe** [reply]
I try to extend the scrollytelling observable framework example with markdownit. For this purpose, I create two new balises : the first to enter the scrolling (::: scroll-container) and the second to describe a step (~~~ scroll-section figure_name).

The figures are stored inside a dictionary. To display a figure in a scroll section, you just need to add the correct key (figure_name) from the dictionary.

The exemple is here : [Examples of scrolling with observablehq framework | Scrollytelling 36](https://francoissemecurbe.observablehq.cloud/scrollytellingwithobservableframework/)

and the index.md here : [scrollytelling_observablehq/scrollytelling/src/index.md at 41dee7853b754519255afb734c9f0b21e4d9a7f3 · fsemecurbe/scrollytelling_observablehq · GitHub 14](https://github.com/fsemecurbe/scrollytelling_observablehq/blob/41dee7853b754519255afb734c9f0b21e4d9a7f3/scrollytelling/src/index.md)

**Comment by cderv** [comment]
Revealjs 5 has a new Scroll View mode which is now available in Quarto 1.6 
See https://prerelease.quarto.org/docs/presentations/revealjs/presenting.html#scroll-view

This could help in some scrollytelling scenario maybe

**Comment by blenback** [comment]
Came to share [Closeread](https://closeread.dev/) but realised that it seems to be the output of @jimjam-slam's experiments mentioned in this discussion? Really awesome stuff!

**Reply by jimjam-slam** [reply]
Oh, I forgot to come back and advertise it 😂 Yeah, @andrewpbray reached out to chat about them and then we went and made Closeread 🥳

**Reply by mcanouil** [reply]
Small note: Better to avoid advertising it (or anything) in all possible discussions/issues.
Show&Tell is there for that.
